BlogApp/utils/types.ts

```diff
@@ -0,0 +1,45 @@
+export interface BlogDetailState {
+  blogId: string;
+  uid: string | null;
```

Comment:
rather than setting data type as string or null . you can make such parameters as optional. so it would be `uid?: string;`